### PR TITLE
[2/2] MNIST Score-Based Generative Modeling (Diffusion Model)

### PR DESCRIPTION
**Pull request by jakee417**
Follow on work to previous [pr](https://github.com/FluxML/model-zoo/pull/349). Ties together `Flux.jl` and `DifferentialEquations.jl` in the context of diffusion models, gives people another option to the typical `PyTorch` + `scipy` combo.

Adds:
- Euler-Maruyama sampler from `DifferentialEquations.jl`
- Probability Flow ODE simulations from `DifferentialEquations.jl`
- Cool plots and animations!

### CODE REVIEW

**Comment by ToucheSir** [comment]
I'm not the right person to review the diffeq part, but the changes look reasonable and the animation is great.

**Comment by jakee417** [comment]
The implementation more or less follows the official examples:

https://diffeq.sciml.ai/stable/tutorials/ode_example/
And 

https://diffeq.sciml.ai/stable/tutorials/sde_example/

File: vision/diffusion_mnist/diffusion_plot.jl [hunk 1]
```diff
@@ -75,19 +99,78 @@ function predictor_corrector_sampler(model, init_x, time_steps, Δt, snr=0.16f0)
...
+"""
+Helper to create a SDEProblem with DifferentialEquations.jl
+
+# Notes
+The reverse-time SDE is given by:  
+𝘥x = -σ²ᵗ 𝘚₀(𝙭, 𝘵)𝘥𝘵 + σᵗ𝘥𝘸  
+⟹ `f(u, p, t)` = -σ²ᵗ 𝘚₀(𝙭, 𝘵)  
+⟹ `g(u, p, t` = σᵗ
+"""
+function DifferentialEquations_problem(model, init_x, time_steps, Δt)
```
Comment (CarloLucibello) [inline vision/diffusion_mnist/diffusion_plot.jl:111]: Is this the right place for this function?

**Comment by CarloLucibello** [comment]
Super cool, thanks!

**Pull request merged**